flang/lib/Lower/ConvertVariable.cpp

```diff
@@ -462,19 +479,15 @@ static void instantiateGlobal(Fortran::lower::AbstractConverter &converter,
   auto globalName = Fortran::lower::mangle::mangleName(sym);
   auto loc = converter.genLocation(sym.name());
   fir::GlobalOp global = builder.getNamedGlobal(globalName);
-  if (!global) {
-    if (var.isDeclaration()) {
-      // Using a global from a module not defined in this compilation unit.
-      mlir::StringAttr externalLinkage;
-      global = declareGlobal(converter, var, globalName, externalLinkage);
-    } else {
-      mlir::StringAttr linkage; // external if remains empty.
-      if (!sym.owner().IsModule() &&
-          !Fortran::semantics::FindCommonBlockContaining(sym))
-        linkage = builder.createInternalLinkage();
-      global = defineGlobal(converter, var, globalName, linkage);
-    }
-  }
+  mlir::StringAttr linkage = getLinkageAttribute(builder, var);
+  if (var.isModuleVariable())
+    // Using a global from a module. If it belongs to this
+    // compilation unit, its definition will be or was added when lowering the
+    // module.
+    global = declareGlobal(converter, var, globalName, linkage);
```

Comment:
nit: I'd use braces with this multiline comment being present, just to make it clear the comment belongs to the `then` arm of the condition, etc.

Reply:
The comment could be formatted to fewer lines, or possibly replaced with a shorter - _A module global was or will be defined when lowering the module._

Reply:
Thanks Val, I used your rephrasing and added braces.